app/grandchallenge/evaluation/forms.py

```diff
@@ -629,9 +617,32 @@ def check_submission_limit_avoidance(self, *, creator):
             self._phase.handle_submission_limit_avoidance(user=creator)
             self.raise_submission_limit_error()
 
+    def save(self, *args, **kwargs):
+        if self._phase.submission_kind == SubmissionKindChoices.ALGORITHM:
+            self.instance.algorithm_requires_gpu_type = self.cleaned_data[
+                "algorithm_image"
+            ].requires_gpu_type
+            self.instance.algorithm_requires_memory_gb = self.cleaned_data[
+                "algorithm_image"
+            ].requires_memory_gb
+        else:
+            self.instance.algorithm_requires_gpu_type = ""
+            self.instance.algorithm_requires_memory_gb = 0
```

Comment:
Not sure that this is the best but these fields are non-editable so cannot be included in the form.

Reply:
I think this is the way to go given that the fields cannot be added to the form.